ui/src/localization.ts

```diff
@@ -148,3 +153,55 @@ export const loadTranslations = () => {
     exhaustiveCheck(language)
   }, [data?.myUserPreferences.language])
 }
+
+export const SetMapLanguages = (map: mapboxgl.Map) => {
+  if (isNil(map_language)) {
+    map.addControl(new MapboxLanguage({ defaultLanguage: 'en' }))
+    return
+  }
+
+  switch (map_language) {
+    case LanguageTranslation.Danish:
```

Comment:
We already have all of this information within the `Userpreferences.tsx` called `languagePreferences`. We should extract this and the elements in `languagePreferences` into a map within this file which is `LanguageTranslation -> languageCode` Although in this case it does appear that the Chinese traditional & simiplified do not use the same code as we do within `UserPreferences`.

Rewriting it in this way gives us somewhere more central to pick up on language codes and means things don't get missed in the future.